gensim/models/word2vec.py

```diff
@@ -529,20 +529,20 @@ def __init__(self, sentences=None, size=100, alpha=0.025, window=5, min_count=5,
         super(Word2Vec, self).__init__(
             sentences=sentences, workers=workers, vector_size=size, epochs=iter, callbacks=callbacks,
             batch_words=batch_words, trim_rule=trim_rule, sg=sg, alpha=alpha, window=window, seed=seed,
-            hs=hs, negative=negative, cbow_mean=cbow_mean, min_alpha=min_alpha, compute_loss=compute_loss,
+            hs=hs, negative=negative, num_bits=num_bits, cbow_mean=cbow_mean, min_alpha=min_alpha, compute_loss=compute_loss,
             fast_version=FAST_VERSION)
 
     def _do_train_job(self, sentences, alpha, inits):
         """
         Train a single batch of sentences. Return 2-tuple `(effective word count after
         ignoring unknown words and sentence length trimming, total word count)`.
         """
-        work, neu1 = inits
+        work1, work2, neu1 = inits
         tally = 0
         if self.sg:
-            tally += train_batch_sg(self, sentences, alpha, work, self.compute_loss)
+            tally += train_batch_sg(self, sentences, alpha, work1, self.compute_loss)
         else:
-            tally += train_batch_cbow(self, sentences, alpha, work, neu1, self.compute_loss)
+            tally += train_batch_cbow(self, sentences, alpha, work1, work2, neu1, self.compute_loss, self.num_bits)
```

Comment:
I think it's better to extract `num_bits` directly in cython from `self` (for save more consistent interfaces), like `cdef int negative = model.negative`